scripts/build/Jenkins/Jenkinsfile

```diff
@@ -461,9 +457,26 @@ def UploadAPLogs(String platformName, String jobName, String workspace, Map para
     }
 }
 
-def PostBuildCommonSteps(String workspace, boolean mount = true) {
-    echo 'Starting post-build common steps...'
+def UploadTestArtifacts(String workspace, String outputDirectory) {
+    catchError(message: "Error archiving test artifacts (this won't fail the build)", buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+        def cmakeBuildDir = [workspace, ENGINE_REPOSITORY_NAME, outputDirectory].join('/')
+        echo "Uploading Test Artifacts: ${cmakeBuildDir}/Testing"
+        ArchiveArtifactsOnS3("${cmakeBuildDir}/Testing", "test_artifacts", true)
+    }
+}
 
+def PostBuildCommonSteps(String workspace, Map params, boolean mount = true) {
+    echo 'Starting post-build common steps...'
+    if (params && params.containsKey('OUTPUT_DIRECTORY')){
+        dir([workspace, ENGINE_REPOSITORY_NAME, params.OUTPUT_DIRECTORY].join('/')){
+            // Clean up Testing directory
+            palRmDir("Testing")
+            // Recreate test runner xml directories that need to be pre generated
+            palMkdir("Testing/Pytest")
+            palMkdir("Testing/Gtest")
```

Comment:
Why do these need to be pregenerated? Won't the test runner create them? Is this working around a permissions issue?

Reply:
This is a workaround for a race condition where two tests running in parallel will attempt to create the folder when outputting xmls

Reply:
Interesting! May be worth clarifying the comment. Let's also track that in another issue, since it sounds like it needs compensation closer to the test-tools.

Reply:
Clarified comment and I will create an issue to track